src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -35,62 +38,51 @@ public RouteValueDictionary()
         /// </remarks>
         public RouteValueDictionary(object values)
         {
-            var otherDictionary = values as RouteValueDictionary;
-            if (otherDictionary != null)
+            RouteValueDictionary dictionary;
```

Comment:
Do we intentionally want to allow null to be passed to this class? The code implies that we do, but i'm wondering if its intentional.